pkg/controllers/argocd_resource_hook_consumer.go

```diff
@@ -0,0 +1,66 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/porter-dev/porter-agent/api/server/types"
+	"github.com/porter-dev/porter-agent/internal/models"
+	"github.com/porter-dev/porter-agent/internal/repository"
+)
+
+// ArgoCDResourceHookConsumer contains all information required to consumer an ArgoCDResourceHook
+type ArgoCDResourceHookConsumer struct {
+	Repository *repository.Repository
+}
+
+// NewArgoCDResourceHookConsumer creates an ArgoCDResourceHookConsumer
+func NewArgoCDResourceHookConsumer(repo *repository.Repository) ArgoCDResourceHookConsumer {
+	return ArgoCDResourceHookConsumer{
+		Repository: repo,
+	}
+}
+
+// Consume contains all business logic for consuming an argo resource hook
+func (co ArgoCDResourceHookConsumer) Consume(ctx context.Context, argoEvent types.ArgoCDResourceHook) error {
+	fmt.Printf("Received argo hook: %#v\n", argoEvent)
```

Comment:
```suggestion
	co.logger.Caller().Info().Msgf("Received argo hook: %#v\n", argoEvent)
```

Might make sense to initialize `ArgoCDResourceHookConsumer` with the shared logger.